src/main/scala/io/pravega/connectors/spark/PravegaSourceProvider.scala

```diff
@@ -1,519 +1,570 @@
+    }
+
+
+    /**
+     * Creates a {@link Batch} to scan the data from this data source.
+     *
+     * This is used to read a Pravega stream as a dataframe in a batch job.
+     *
+     * Unbounded stream cuts (earliest, latest, unbounded) are bound only once.
+     * Late binding is not available.
+     *
+     * If this method fails (by throwing an exception), the action will fail and no Spark job will be
+     * submitted.
+     *
+     */
+    override def toBatch(): Batch = {
+
+      val caseInsensitiveParams = CaseInsensitiveMap(options.asScala.toMap)
+      validateBatchOptions(caseInsensitiveParams)
+      val clientConfig = buildClientConfig(caseInsensitiveParams)
+      val scopeName = caseInsensitiveParams(PravegaSourceProvider.SCOPE_OPTION_KEY)
+      val streamName = caseInsensitiveParams(PravegaSourceProvider.STREAM_OPTION_KEY)
+
+      val metadataTableName = caseInsensitiveParams.get(PravegaSourceProvider.METADATA_OPTION_KEY).map(MetadataTableName.withName)
+      if (metadataTableName.isDefined) {
+        log.info("in metadataTableName")
+        return createMetadataReader(
+          scopeName,
+          streamName,
+          clientConfig,
+          metadataTableName.get,
+          caseInsensitiveParams)
+      }
+
+      val startStreamCut = PravegaSourceProvider.getPravegaStreamCut(
+        caseInsensitiveParams, PravegaSourceProvider.START_STREAM_CUT_OPTION_KEY, EarliestStreamCut)
+
+      val endStreamCut = PravegaSourceProvider.getPravegaStreamCut(
+        caseInsensitiveParams, PravegaSourceProvider.END_STREAM_CUT_OPTION_KEY, LatestStreamCut)
+
+      log.info(s"createReader: clientConfig=${clientConfig}, scopeName=${scopeName}, streamName=${streamName}"
+        + s" startStreamCut=${startStreamCut}, endStreamCut=${endStreamCut}")
+
+      createStreams(caseInsensitiveParams)
+
+      new PravegaBatch(
+        scopeName,
+        streamName,
+        clientConfig,
+        caseInsensitiveParams,
+        startStreamCut,
+        endStreamCut)
+    }
+
+
+    /**
+     * Creates a {@link MicroBatchStream} to read batches of data from this data source in a
+     * streaming query.
+     * This is used to read a Pravega stream as a datastream in a structured streaming job.
+     *
+     * The execution engine will create a micro-batch reader at the start of a streaming query,
+     * alternate calls to setOffsetRange and planInputPartitions for each batch to process, and
+     * then call stop() when the execution is complete. Note that a single query may have multiple
+     * executions due to restart or failure recovery.
+     *
+     * @param checkpointLocation a path to Hadoop FS scratch space that can be used for failure
+     *                           recovery. Readers for the same logical source in the same query
+     *                           will be given the same checkpointLocation.
+     */
+    override def toMicroBatchStream(checkpointLocation: String): MicroBatchStream = {

```

Comment:
Add a comment explaining why checkpointLocation is not used this by this connector.